apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -41,28 +41,62 @@ public SpannerParameter()
         /// </summary>
         /// <param name="spannerColumnName"></param>
         /// <param name="type"></param>
+        /// <param name="value"></param>
         /// <param name="sourceColumn"></param>
         /// <param name="size"></param>
-        public SpannerParameter(string spannerColumnName, SpannerDbType type, string sourceColumn = null, int size = 0)
+        public SpannerParameter(string spannerColumnName, SpannerDbType type, object value = null,
+                string sourceColumn = null, int size = 0)
         {
             _spannerColumnName = spannerColumnName;
-            TypeCode = type.GetSpannerTypeCode();
+            SpannerDbType = type ?? SpannerDbType.Unspecified;
+            Value = value;
             SourceColumn = sourceColumn;
             Size = size;
         }
 
         /// <inheritdoc />
         public override DbType DbType
         {
-            get { return TypeCode.GetDbType(); }
-            set { TypeCode = value.GetSpannerType(); }
+            get => (SpannerDbType?.TypeCode.GetDbType()).GetValueOrDefault(DbType.Object);
+            set
+            {
+                switch (value)
+                {
+                    case DbType.Binary:
+                        SpannerDbType = SpannerDbType.Bytes;
+                        break;
+                    case DbType.Boolean:
+                        SpannerDbType = SpannerDbType.Bool;
+                        break;
+                    case DbType.Date:
+                        SpannerDbType = SpannerDbType.Date;
+                        break;
+                    case DbType.DateTime:
+                        SpannerDbType = SpannerDbType.Timestamp;
+                        break;
+                    case DbType.Double:
+                        SpannerDbType = SpannerDbType.Float64;
+                        break;
+                    case DbType.Int64:
+                        SpannerDbType = SpannerDbType.Int64;
+                        break;
+                    case DbType.Object:
+                        SpannerDbType = SpannerDbType.Unspecified;
+                        break;
+                    case DbType.String:
+                        SpannerDbType = SpannerDbType.String;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(DbType), value, null);
+                }
+            }
         }
 
         /// <inheritdoc />
         public override ParameterDirection Direction
         {
-            get { return ParameterDirection.Input; }
-            set { throw new InvalidOperationException("Spanner only supports input parameters."); }
+            get => ParameterDirection.Input;
+            set => throw new InvalidOperationException("Spanner only supports input parameters.");
```

Comment:
This will throw on being set to `Direction.Input`, which is presumably valid. Is this desirable?